### PR TITLE
[Pre-delivery] Feature/0296 possibility to update video streaming capabilities

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Implements https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md

This PR is **[ready]** for review.

### Risk
This PR makes **[minor]** API changes.

### Testing Plan
http://172.26.147.12/view/0296_update_video_streaming_capabilities/

### Summary

1. New additionalVideoStreamingCapabilities parameter of VideoStreamingCapability processing
- Store additionalVideoStreamingCapabilities from UI.GetCapabilities response
- Send additionalVideoStreamingCapabilities to the mobile app via GetSystemCapabilities
- Resend additionalVideoStreamingCapabilities to the mobile app via OnSystemCapabilityUpdated notification from HMI
2. OnSystemCapabilityUpdated with videoStreamingCapabilities processing
- The notification from HMI which contains appID parameter must be re-sent to appropriate mobile application only if it had been subscribed on it else it must be ignored
- The notification from HMI which does not contain appID parameter must be ignored
3. OnAppCapabilityUpdated notification processing 
- The notification from the mobile must be re-sent to the HMI

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_bc_system_capability_updated_notification_from_hmi.cc [hunk 1]
```diff
@@ -36,6 +36,7 @@
 #include "application_manager/message_helper.h"
 #include "interfaces/HMI_API.h"
 #include "interfaces/MOBILE_API.h"
+#include "sdl_rpc_plugin/extensions/system_capability_app_extension.h"
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_bc_system_capability_updated_notification_from_hmi.cc:39]: Please remove sdl_rpc_plugin/, and ajust cmake to use "extensions/system_capability_app_extension.h"

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_bc_system_capability_updated_notification_from_hmi.cc [hunk 2]
```diff
@@ -135,6 +145,48 @@ void OnBCSystemCapabilityUpdatedNotificationFromHMI::Run() {
       }
       break;
     }
+    case mobile_apis::SystemCapabilityType::VIDEO_STREAMING: {
+      if (!system_capability.keyExists(strings::video_streaming_capability)) {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_bc_system_capability_updated_notification_from_hmi.cc:149]: extract it from switch to function

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/on_bc_system_capability_updated_notification_from_hmi_test.cc [hunk 3]
```diff
@@ -47,15 +49,34 @@ using ::testing::Return;
...
 
 MATCHER(CheckMessageToMobile, "") {
   const auto function_id = mobile_apis::FunctionID::OnSystemCapabilityUpdatedID;
 
+  const bool is_function_id_matched =
+      function_id == static_cast<am::mobile_api::FunctionID::eType>(
+                         (*arg)[strings::params][strings::function_id].asInt());
+  const bool app_id_exist =
+      (*arg)[strings::msg_params].keyExists(strings::app_id);
+  bool is_connection_key_correct = true;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/on_bc_system_capability_updated_notification_from_hmi_test.cc:68]: make a lambda instead bool

**Comment by YarikMamykin** [comment]
@LitvinenkoIra please don't forget about https://github.com/LuxoftSDL/sdl_core/pull/78

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/bc_on_app_capability_updated_notification.h [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_RPC_PLUGINS_SDL_RPC_PLUGIN_INCLUDE_SDL_RPC_PLUGIN_COMMANDS_HMI_ON_APP_CAPABILITY_UPDATED_NOTIFICATION_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_RPC_PLUGINS_SDL_RPC_PLUGIN_INCLUDE_SDL_RPC_PLUGIN_COMMANDS_HMI_ON_APP_CAPABILITY_UPDATED_NOTIFICATION_H_
+
+#include "application_manager/commands/notification_to_hmi.h"
+
+namespace sdl_rpc_plugin {
+namespace app_mngr = application_manager;
```
[the remark this answers is not in the document]
Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/bc_on_app_capability_updated_notification.h:39]: @YarikMamykin I'm not sure if it matters, I think It was just a copy-paste from the other commands classes

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_bc_system_capability_updated_notification_from_hmi.cc (hunk 1 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_bc_system_capability_updated_notification_from_hmi.cc]: @LuxoftAKutsan Done in bbdf770e46270a5cc6f246d411401e8e0e8bfc4e

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_bc_system_capability_updated_notification_from_hmi.cc (hunk 2 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_bc_system_capability_updated_notification_from_hmi.cc]: @LuxoftAKutsan Done in bbdf770e46270a5cc6f246d411401e8e0e8bfc4e

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/on_bc_system_capability_updated_notification_from_hmi_test.cc (hunk 3 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/on_bc_system_capability_updated_notification_from_hmi_test.cc]: @LuxoftAKutsan Done in 6e05066361300834d09b5391052d19c305515d0e

**Comment by LitvinenkoIra** [comment]
@YarikMamykin @LuxoftAKutsan @SNiukalov please check the https://github.com/LuxoftSDL/sdl_core/pull/77/commits/359d5969bdc8018c57ba10fcab60fe69a5cacedd 
The _LoadCapabilitiesFromFile_CheckVideoStreamingCapability_SuccessLoadAndConvert_ unit test was updated according to the changes in the default hmi_capabilities

**Pull request closed without merging**